samples/glean/src/main/java/org/mozilla/samples/glean/GleanApplication.kt

```diff
@@ -6,51 +6,66 @@ package org.mozilla.samples.glean
 import mozilla.components.service.experiments.Experiments
+import mozilla.components.service.glean.RustHttpConfig
 import mozilla.components.support.base.log.Log
 import mozilla.components.support.base.log.sink.AndroidLogSink
 import org.mozilla.samples.glean.GleanMetrics.Basic
 import org.mozilla.samples.glean.GleanMetrics.Test
 import org.mozilla.samples.glean.GleanMetrics.Custom
 import org.mozilla.samples.glean.GleanMetrics.Pings
+import kotlin.coroutines.CoroutineContext
+
 
 class GleanApplication : Application() {
 
     override fun onCreate() {
-        super.onCreate()
-
-        // We want the log messages of all builds to go to Android logcat
-        Log.addSink(AndroidLogSink())
-
-        // Register the sample application's custom pings.
-        Glean.registerPings(Pings)
-
-        // Initialize the Glean library. Ideally, this is the first thing that
-        // must be done right after enabling logging.
-        val client by lazy { HttpURLConnectionClient() }
-        val httpClient = ConceptFetchHttpUploader.fromClient(client)
-        val config = Configuration(httpClient = httpClient)
-        Glean.initialize(applicationContext, uploadEnabled = true, configuration = config)
-
-        // Initialize the Experiments library and pass in the callback that will generate a
-        // broadcast Intent to signal the application that experiments have been updated. This is
-        // only relevant to the experiments library, aside from recording the experiment in Glean.
-        Experiments.initialize(applicationContext, ExperimentsConfig(httpClient = client)) {
-            val intent = Intent()
-            intent.action = "org.mozilla.samples.glean.experiments.updated"
-            sendBroadcast(intent)
-        }
+            super.onCreate()
+            // We want the log messages of all builds to go to Android logcat
+            Log.addSink(AndroidLogSink())
+
+            // Register the sample application's custom pings.
+            Glean.registerPings(Pings)
 
-        Test.timespan.start()
+            // Initialize the Glean library. Ideally, this is the first thing that
+            // must be done right after enabling logging.
+            val client by lazy { HttpURLConnectionClient() }
+            val httpClient = ConceptFetchHttpUploader.fromClient(client)
+            val config = Configuration(httpClient = httpClient)
+            Glean.initialize(applicationContext, uploadEnabled = true, configuration = config)
+            RustHttpConfig.setClient(lazy { HttpURLConnectionClient() })
+            Thread {
+                GlobalScope.launch {
+                    var exp = mozilla.components.service.glean.Experiments
+                    exp.initialize("https://kinto.dev.mozaws.net/v1/",
+                            "messaging-collection",
+                            "main",
+                            applicationContext.dataDir.path)
+                    val res = exp.getBranch("button-color")
```

Comment:
a `TODO` here for us would be something like:

* Send an enrolled event for the experiment
* Send a saw `button-color` button. AKA "User saw some branch of the experiment"